src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/JsonFunctionCallUtilities.cs

```diff
@@ -74,179 +103,141 @@ internal static partial class FunctionCallHelpers
     public static JsonElement InferParameterJsonSchema(
         AIFunctionParameterMetadata parameterMetadata,
         AIFunctionMetadata functionMetadata,
-        JsonSerializerOptions? options)
+        JsonSerializerOptions? serializerOptions,
+        JsonSchemaInferenceOptions? inferenceOptions = null)
     {
-        options ??= functionMetadata.JsonSerializerOptions;
+        _ = Throw.IfNull(parameterMetadata);
+        _ = Throw.IfNull(functionMetadata);
 
-        if (ReferenceEquals(options, functionMetadata.JsonSerializerOptions) &&
+        serializerOptions ??= functionMetadata.JsonSerializerOptions;
+
+        if (ReferenceEquals(serializerOptions, functionMetadata.JsonSerializerOptions) &&
             parameterMetadata.Schema is JsonElement schema)
         {
             // If the resolved options matches that of the function metadata,
             // we can just return the precomputed JSON schema value.
             return schema;
         }
 
-        if (options is null)
+        if (serializerOptions is null)
         {
-            return TrueJsonSchema;
+            return _trueJsonSchema;
         }
 
         return InferParameterJsonSchema(
             parameterMetadata.ParameterType,
             parameterMetadata.Name,
-            parameterMetadata.Description,
-            parameterMetadata.HasDefaultValue,
-            parameterMetadata.DefaultValue,
-            options);
+            serializerOptions,
+            description: parameterMetadata.Description,
+            hasDefaultValue: parameterMetadata.HasDefaultValue,
+            defaultValue: parameterMetadata.DefaultValue,
+            inferenceOptions);
     }
 
     /// <summary>
     /// Determines a JSON schema for the provided parameter metadata.
     /// </summary>
     /// <param name="type">The type of the parameter.</param>
-    /// <param name="name">The name of the parameter.</param>
+    /// <param name="parameterName">The name of the parameter.</param>
+    /// <param name="serializerOptions">The options used to extract the schema from the specified type.</param>
     /// <param name="description">The description of the parameter.</param>
     /// <param name="hasDefaultValue">Whether the parameter is optional.</param>
     /// <param name="defaultValue">The default value of the optional parameter, if applicable.</param>
-    /// <param name="options">The options used to extract the schema from the specified type.</param>
+    /// <param name="inferenceOptions">The options controlling schema inference.</param>
     /// <returns>A JSON schema document encoded as a <see cref="JsonElement"/>.</returns>
     public static JsonElement InferParameterJsonSchema(
         Type? type,
-        string name,
-        string? description,
-        bool hasDefaultValue,
-        object? defaultValue,
-        JsonSerializerOptions options)
+        string parameterName,
+        JsonSerializerOptions serializerOptions,
```

Comment:
Here as well, I'd have expected serializerOptions to be optional and to be at the end of the parameter list. It feels strange to me coming between parameterName and other things about the parameter, like description and default value.

Reply:
We can't make the JSO optional in this case since there's no `AIFunctionParameterMetadata` to tap into. We could of course use the same technique as `JsonDefaults` does, but that's defined in a separate assembly and it didn't make a lot of sense to try and duplicate in primitives as well. This overload is only intended for use by `AIFunctionFactory` or similar components but not chat service implementations the use the first overload.

Reply:
> but that's defined in a separate assembly and it didn't make a lot of sense to try and duplicate in primitives as well.

Why not?

Reply:
Should we expose from JsonFunctionCallUtilities the default JSO we use?

Reply:
Done.